python/lsst/alert/packet/schema.py

```diff
@@ -24,17 +24,55 @@
 
 import io
 import os.path
+import pkg_resources
 
 import fastavro
 
-__all__ = ["get_schema_root", "Schema"]
+__all__ = ["get_schema_root", "get_latest_schema_version", "get_schema_path", "Schema"]
 
 def get_schema_root():
     """Return the root of the directory within which schemas are stored.
     """
-    return os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../../../schema"))
+    return pkg_resources.resource_filename(__name__, "schema")
+
+def get_latest_schema_version():
+    """Get the latest schema version..
+
+    Returns
+    -------
+    major : `int`
+        The major version number.
+    minor : `int`
+        The minor version number.
+    """
+    val = pkg_resources.resource_string(__name__, "schema/latest.txt")
+    clean = val.strip()
+    major, minor = clean.split(b".", 1)
+    return int(major), int(minor)
+
+def get_schema_path(major, minor):
+    """Get the path to a package resource directory housing alert schema
+    definitions
+
+    Parameters
+    ----------
+    major : `int`
+        Major version number for the schema.
+    minor : `int`
+        Minor version number for the schema.
+
+    Returns
+    -------
+    path : `str`
+        Path to the directory containing the schemas.
+    """
+
+    path = "/".join(["schema", str(major), str(minor)])
```

Comment:
If I were @timj , I'd probably mutter something about `os.path.join` here...

Reply:
This is for the same reason as discussed in https://github.com/lsst/alert_packet/pull/15#discussion_r434238043. Resource paths _look_ like OS paths, but they aren't actually the same thing. They always use slashes, so os.path.join would be wrong here (see https://setuptools.readthedocs.io/en/latest/pkg_resources.html#basic-resource-access).

Yeah, it's weird.